Mathlib/NumberTheory/LSeries/HurwitzZetaEven.lean

```diff
@@ -395,3 +407,398 @@ lemma completedCosZeta₀_one_sub (a : UnitAddCircle) (s : ℂ) :
+    rwa [div_ofNat_re, div_lt_div_right two_pos]]
+  refine (hasSum_mellin_pi_mul_sq (zero_lt_one.trans hs) hF ?_).congr_fun fun n ↦ ?_
+  · apply (((summable_one_div_int_add_rpow 0 s.re).mpr hs).div_const 2).of_norm_bounded
+    intro i
+    simp only [c, (by { push_cast; ring } : 2 * π * I * a * i = ↑(2 * π * a * i) * I), norm_div,
+      RCLike.norm_ofNat, norm_norm, Complex.norm_exp_ofReal_mul_I, add_zero, norm_one,
+      norm_of_nonneg (by positivity : 0 ≤ |(i : ℝ)| ^ s.re), div_right_comm, le_rfl]
+  · simp only [c, Int.cast_eq_zero, ← Int.cast_abs, ofReal_intCast, div_right_comm, mul_div_assoc]
+
+/-- Formula for `completedCosZeta` as a Dirichlet series in the convergence range
+(second version, with sum over `ℕ`). -/
+lemma hasSum_nat_completedCosZeta (a : ℝ) {s : ℂ} (hs : 1 < re s) :
+    HasSum (fun n : ℕ ↦ if n = 0 then 0 else Gammaℝ s * Real.cos (2 * π * a * n) / (n : ℂ) ^ s)
+    (completedCosZeta a s) := by
+  have aux : ((|0| : ℤ) : ℂ) ^ s = 0 := by
+    rw [abs_zero, Int.cast_zero, zero_cpow (ne_zero_of_one_lt_re hs)]
+  have hint := (hasSum_int_completedCosZeta a hs).sum_nat_of_sum_int
+  rw [aux, div_zero, zero_div, add_zero] at hint
+  refine hint.congr_fun fun n ↦ ?_
+  split_ifs with h
+  · simp only [h, Nat.cast_zero, aux, div_zero, zero_div, neg_zero, zero_add]
+  · simp only [ofReal_cos, ofReal_mul, ofReal_ofNat, ofReal_natCast, Complex.cos,
+      show 2 * π * a * n * I = 2 * π * I * a * n by ring, neg_mul, mul_div_assoc,
+      div_right_comm _ (2 : ℂ), Int.cast_natCast, Nat.abs_cast, Int.cast_neg, mul_neg, abs_neg, ←
+      mul_add, ← add_div]
+
+/-- Formula for `completedHurwitzZetaEven` as a Dirichlet series in the convergence range. -/
+lemma hasSum_int_completedHurwitzZetaEven (a : ℝ) {s : ℂ} (hs : 1 < re s) :
+    HasSum (fun n : ℤ ↦ Gammaℝ s / (↑|n + a| : ℂ) ^ s / 2) (completedHurwitzZetaEven a s) := by
+  have hF (t : ℝ) (ht : 0 < t) : HasSum (fun n : ℤ ↦ if n + a = 0 then 0
+      else (1 / 2 : ℂ) * rexp (-π * (n + a) ^ 2 * t)) ((evenKernel a t - _) / 2) := by
+    refine (ofReal_sub .. ▸ (hasSum_ofReal.mpr (hasSum_int_evenKernel₀ a ht)).div_const
+      2).congr_fun fun n ↦ ?_
+    split_ifs
+    · rw [ofReal_zero, zero_div]
+    · rw [mul_comm, mul_one_div]
+  rw [show completedHurwitzZetaEven a s = mellin (fun t ↦ ((evenKernel (↑a) t : ℂ) -
+        ↑(if (a : UnitAddCircle) = 0 then 1 else 0 : ℝ)) / 2) (s / 2) by
+    simp_rw [mellin_div_const, apply_ite ofReal', ofReal_one, ofReal_zero]
+    refine congr_arg (· / 2) ((hurwitzEvenFEPair a).hasMellin (?_ : 1 / 2 < (s / 2).re)).2.symm
+    rwa [div_ofNat_re, div_lt_div_right two_pos]]
+  refine (hasSum_mellin_pi_mul_sq (zero_lt_one.trans hs) hF ?_).congr_fun fun n ↦ ?_
+  · simp_rw [← mul_one_div ‖_‖]
+    apply Summable.mul_left
+    rwa [summable_one_div_int_add_rpow]
+  · rw [mul_one_div, div_right_comm]
+
+/-!
+## The un-completed even Hurwitz zeta
+-/
+
+/-- Technical lemma which will give us differentiability of Riemann zeta at `s = 0`. -/
```

Comment:
Are you planning to apply this to streamline something about Riemann zeta, or should one just change the docstring by replacing Riemann with Hurwitz?

Reply:
Good point, that was unclear, I'll change it.